src/libs/actions/Report.js

```diff
@@ -762,8 +763,8 @@ function fetchAllReports(
             }
 
             const firstReportID = _.first(reportIDs);
-            const currentReportID = firstReportID ? String(firstReportID) : '';
-            Onyx.merge(ONYXKEYS.CURRENTLY_VIEWED_REPORTID, currentReportID);
+            // eslint-disable-next-line no-use-before-define
+            updateCurrentlyViewedReportID(firstReportID);
```

Comment:
Maybe I should move `updateCurrentlyViewedReportID` to be declared before this usage so the eslint override can be removed. 

About that eslint rule it can be tweaked. There's nothing wrong using a `function` before it's defined and the rule allows for such configuration. The problem is when you try to use a `class` or a variable before they are defined.
This style rule is forcing specifics to be defined at the top, which usually degrades readability.

Reply:
>About that eslint rule it can be tweaked.

It has been brought up before. We generally agree with the reasoning laid out here 

https://eslint.org/docs/rules/no-use-before-define